### PR TITLE
Fix entire project + Kafka + Sessions Service

### PR DESCRIPTION
**Pull request by jejikeh**
Fixed problems with the code style, refactored the identity service, simplified work with the JWT token. Made a normal data seeder. 
Remove weird error handling in product service.

### CODE REVIEW

File: src/Versity.ApiGateway/Versity.ApiGateway/Extensions/ProgramExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    public static WebApplicationBuilder ConfigureBuilder(this WebApplicationBuilder builder)
+    {
+        builder.Configuration
+            .SetBasePath(builder.Environment.ContentRootPath)
+            .AddJsonFile("ocelot.json", optional: false, reloadOnChange: true)
+            .AddEnvironmentVariables();
+        
+        builder.Services.AddOcelot(builder.Configuration);
+        builder.Services.AddJwtAuthentication(builder.Configuration);
+        builder.Services.AddCors(options => options.AddPolicy("AllowAll", policy =>
```
Comment (KatsiarynaAnashkevich) [inline src/Versity.ApiGateway/Versity.ApiGateway/Extensions/ProgramExtensions.cs]: Extension is needed

File: src/Versity.Products/Core/Application/RequestHandlers/Commands/CreateProduct/CreateProductCommandHandler.cs [hunk 2]
```diff
@@ -14,13 +14,14 @@ public CreateProductCommandHandler(IVersityProductsRepository products)
         _products = products;
     }
 
-    public async Task<Result<Product>> Handle(CreateProductCommand request, CancellationToken cancellationToken)
+    public async Task<Product> Handle(CreateProductCommand request, CancellationToken cancellationToken)
     {
         var productId = Guid.NewGuid();
         while (await _products.GetProductByIdAsync(productId, cancellationToken) is not null)
```
Comment (KatsiarynaAnashkevich) [inline src/Versity.Products/Core/Application/RequestHandlers/Commands/CreateProduct/CreateProductCommandHandler.cs]: Why while is here?

File: src/Versity.Products/Core/Application/RequestHandlers/Queries/GetAllProducts/GetAllProductsQueryHandler.cs [hunk 3]
```diff
@@ -23,6 +22,6 @@ public Task<Result<IEnumerable<Product>>> Handle(GetAllProductsQuery request, Ca
             .Take(10)
             .ToList();
 
-        return Task.Run(() => Result.Ok((IEnumerable<Product>)products), cancellationToken);
+        return Task.Run(() => (IEnumerable<Product>)products, cancellationToken);
```
Comment (KatsiarynaAnashkevich) [inline src/Versity.Products/Core/Application/RequestHandlers/Queries/GetAllProducts/GetAllProductsQueryHandler.cs]: Why?

File: src/Versity.Products/Core/Application/RequestHandlers/Queries/GetAllProducts/GetAllProductsQueryHandler.cs [hunk 4]
```diff
@@ -23,6 +22,6 @@ public Task<Result<IEnumerable<Product>>> Handle(GetAllProductsQuery request, Ca
             .Take(10)
```
Comment (KatsiarynaAnashkevich) [inline src/Versity.Products/Core/Application/RequestHandlers/Queries/GetAllProducts/GetAllProductsQueryHandler.cs]: I recommend to create class with constants and use it instesd of values

File: src/Versity.Sessions/Core/Application/Exceptions/ValidationExceptionWithStatusCode.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+﻿using Microsoft.AspNetCore.Http;
+
+namespace Application.Exceptions;
+
+public class ValidationExceptionWithStatusCode : ExceptionWithStatusCode
+{
+    public ValidationExceptionWithStatusCode(object? value = null) : base(StatusCodes.Status403Forbidden, value)
```
Comment (KatsiarynaAnashkevich) [inline src/Versity.Sessions/Core/Application/Exceptions/ValidationExceptionWithStatusCode.cs]: Object? Why object?

File: src/Versity.Sessions/Core/Application/RequestHandlers/Sessions/Commands/CreateSession/CreateSessionCommandHandler.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+    public async Task<Session> Handle(CreateSessionCommand request, CancellationToken cancellationToken)
+    {
+        if (!await _users.IsUserExistAsync(request.UserId))
+        {
+            throw new NotFoundExceptionWithStatusCode("User with specified Id doesnt exist!");
+        }
+        
+        var sessionId = Guid.NewGuid();
+        while (await _sessions.GetSessionByIdAsync(sessionId, cancellationToken) is not null)
```
Comment (KatsiarynaAnashkevich) [inline src/Versity.Sessions/Core/Application/RequestHandlers/Sessions/Commands/CreateSession/CreateSessionCommandHandler.cs]: ?

File: src/Versity.Sessions/Core/Application/RequestHandlers/Sessions/Queries/GetAllSessions/GetAllSessionsQueryHandler.cs [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+    public GetAllSessionsQueryHandler(ISessionsRepository sessions)
+    {
+        _sessions = sessions;
+    }
+
+    public Task<IEnumerable<Session>> Handle(GetAllSessionsQuery request, CancellationToken cancellationToken)
+    {
+        var sessions = _sessions
+            .GetAllSessions()
```
Comment (KatsiarynaAnashkevich) [inline src/Versity.Sessions/Core/Application/RequestHandlers/Sessions/Queries/GetAllSessions/GetAllSessionsQueryHandler.cs:20]: What collection do you get here?

File: src/Versity.Sessions/External/Presentation/Services/GrpcUsersDataService.cs [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+        try
+        {
+            var reply = await client.GetUserRolesAsync(request);
+            
+            return reply.Roles.ToList();
+        }
+        catch (Exception e)
+        {
+            Console.WriteLine($"--> Could`t call to GRPC server {e.Message}");
```
Comment (KatsiarynaAnashkevich) [inline src/Versity.Sessions/External/Presentation/Services/GrpcUsersDataService.cs]: logs

File: src/Versity.Sessions/External/Presentation/Services/GrpcUsersDataService.cs [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    private readonly IConfiguration _configuration;
+
+    public GrpcUsersDataService(IConfiguration configuration)
+    {
+        _configuration = configuration;
+    }
+
+    public async Task<IEnumerable<string>> GetUserRolesAsync(string userId)
+    {
+        var channel = GrpcChannel.ForAddress(_configuration["GrpcUsers"]);
```
Comment (KatsiarynaAnashkevich) [inline src/Versity.Sessions/External/Presentation/Services/GrpcUsersDataService.cs]: And maybe move it to private fields and define in the constructor

File: src/Versity.Users/External/Infrastructure/Persistence/VersityUsersDbContext.cs [hunk 10]
```diff
@@ -10,6 +11,15 @@ public class VersityUsersDbContext : IdentityDbContext<VersityUser>
...
     public VersityUsersDbContext(DbContextOptions options) : base(options)
     {
-        
+    }
+    
+    protected override void OnModelCreating(ModelBuilder modelBuilder)
+    {
+        base.OnModelCreating(modelBuilder);
+        modelBuilder
+            .ApplyConfiguration(new RefreshTokensConfiguration())
```
Comment (KatsiarynaAnashkevich) [inline src/Versity.Users/External/Infrastructure/Persistence/VersityUsersDbContext.cs]: you can apply them in one line

File: src/Versity.Users/External/Presentation/Controllers/UsersController.cs [hunk 11]
```diff
@@ -54,6 +46,17 @@ public async Task<IActionResult> ChangeUserPassword(string id, ChangeUserPasswor
             changeUserPasswordDto.NewPassword,
             id);
         var result = await Sender.Send(command, cancellationToken);
+        
         return Ok(result);
     }
+    
+    [Authorize(Roles = "Admin")]
+    [HttpPost("setadmin/{id:guid}")]
```
Comment (KatsiarynaAnashkevich) [inline src/Versity.Users/External/Presentation/Controllers/UsersController.cs]: bad route name

**Review by KatsiarynaAnashkevich — COMMENTED** [review]
Why do you change command/query to request?

**Comment by jejikeh** [comment]
I used ICommand and IQuery when experimenting with the FluentResults library. 

But then I decided to remove it because I didn't really like handling errors in their way. 

And I Command/Query seems useless to me, since I have already separated the command and queries in different namespaces and folders.

**Comment by jejikeh** [comment]
The PR is quite big, but in overall this is refactoring after the review

**Review by KatsiarynaAnashkevich — APPROVED** [review]

**Pull request merged**